libethereum/BlockChain.cpp

```diff
@@ -268,7 +276,7 @@ unsigned BlockChain::open(fs::path const& _path, WithExisting _we)
         throw;
     }
 
-    if (_we != WithExisting::Verify && !details(m_genesisHash))
+    if (_we != WithExisting::Verify && !rebuildNeeded && !details(m_genesisHash))
```

Comment:
If do this insertion here in case `rebuildNeeded == true`, maybe then we don't need to duplicate this code in `rebuild` ?

Reply:
Ah I see, rebuild will first rename this database anyway